src/fides/core/config/__init__.py

```diff
@@ -93,18 +93,22 @@ def log_all_config_values(self) -> None:
                 )
 
 
-def censor_config(config: FidesConfig) -> Dict[str, Any]:
+def censor_config(config: Union[FidesConfig, Dict[str, Any]]) -> Dict[str, Any]:
     """
     Returns a config that is safe to expose over the API. This function will
     strip out any keys not specified in the `CONFIG_KEY_ALLOWLIST` above.
     """
-    as_dict = config.dict()
+    if not isinstance(config, Dict):
+        as_dict = config.dict()
+    else:
+        as_dict = config
     filtered: Dict[str, Any] = {}
     for key, value in CONFIG_KEY_ALLOWLIST.items():
-        data = as_dict[key]
-        filtered[key] = {}
-        for field in value:
-            filtered[key][field] = data[field]
+        if key in as_dict:
+            data = as_dict[key]
+            filtered[key] = {}
+            for field in value:
+                filtered[key][field] = data[field]
```

Comment:
I think this would break if you only set one of the subsettings - for example, I had only updated notifications.notification_service_type via the API:
```json
{"notifications": {
    "notification_service_type": "twilio_email"
}}
```

You might add another check, like:
```
                if field in data:
                    filtered[key][field] = data[field]
```
or look into defaultdict, I use that too.

Reply:
fixed in https://github.com/ethyca/fides/pull/2551/commits/48f0394497cb0a6ad9717a5a128080aff775db0c